sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -16,37 +16,61 @@
  */
 public final class Utility {
     private static final ClientLogger LOGGER = new ClientLogger(Utility.class);
+    // using 4K as default buffer size: https://stackoverflow.com/a/237495/1473510
+    private static final int BYTE_BUFFER_CHUNK_SIZE = 4096;
 
     private Utility() {
     }
 
     /**
-     * A utility method for converting the input stream to Flux of ByteBuffer.
+     * Creates a Flux of ByteBuffer, with each ByteBuffer wrapping bytes read from the given
+     * InputStream.
      *
-     * @param data The input data which needs to convert to ByteBuffer.
-     *
-     * @return {@link ByteBuffer} which contains the input data.
-     * @throws RuntimeException When I/O error occurs.
+     * @param inputStream InputStream to back the Flux
+     * @return Flux of ByteBuffer backed by the InputStream
      */
-    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data) {
-        return Flux.just(toByteArray(data))
-            .doOnError(error -> LOGGER.warning("Failed to convert stream to byte array - {}", error));
+    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
```

Comment:
Should we consider moving this to core?
cc: @srnagar

Reply:
FYI, I talked to @alzimmermsft  a couple days ago. I will finish this implementation in Form Recognizer first, and then he will decide to move this to core or not.

Reply:
This is something we'll want to add into `FluxUtil` eventually, mind filing an issue for this.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/10337